worker/producer.go

```diff
@@ -15,8 +17,25 @@ type Producer struct {
 	quit          chan chan error
 }
 
-func NewProducer(queue *convoyRedis.RedisQueue) *Producer {
-	consumer := queue.Consumer()
+func NewProducer(cfg config.Configuration, eventQueue queue.Queuer) *Producer {
+	var consumer taskq.QueueConsumer
+	var queue queue.Queuer
+
+	if cfg.Queue.Type == config.RedisQueueProvider {
+		if queue, ok := eventQueue.(*convoyRedis.RedisQueue); ok {
+			consumer = queue.Consumer()
+		}
+	}
+
+	if cfg.Queue.Type == config.InMemoryQueueProvider {
+		if queue, ok := eventQueue.(*convoyMemqueue.MemQueue); ok {
+			consumer = queue.Consumer()
+			err := consumer.Stop()
+			if err != nil {
+				log.Fatal(err)
```

Comment:
why are we stopping the consumer?

Reply:
hm, we stop it here. only to start it again somewhere else?

Reply:
we start it as usual in cmd/server.go.